### PR TITLE
[WIP]Automating BZ #1699943 Service retire test for non-admin user

### PR DESCRIPTION
**Pull request by niyazRedhat**
{{pytest: cfme/tests/ssui/test_ssui_myservice.py -k ' test_retire_service_ssui_user or test_myservice_crud or test_retire_service_ssui or test_service_start' -vv --long-running}}

### CODE REVIEW

**Comment by niyazRedhat** [comment]
@digitronik I am using code from below PR as you are not working on that PR. I need this in current test automation.

https://github.com/ManageIQ/integration_tests/pull/9034

File: cfme/configure/access_control/__init__.py [hunk 1]
```diff
@@ -1107,8 +1107,11 @@ def delete(self, cancel=True):
             else:
                 view = self.create_view(DetailsRoleView, wait=10)  # implicit assert
 
-    def copy(self, name=None):
+    def copy(self, name=None, restriction=None):
         """ Creates copy of existing role
+        Args:
+            name: new name for copied role
+            restriction: Access Restriction for Services, VMs, and Templates
```
Comment (kedark3) [inline cfme/configure/access_control/__init__.py]: **SUGGESTION:** is it a string parameter? Can you please mention it ?

File: cfme/configure/access_control/__init__.py [hunk 2]
```diff
@@ -1118,7 +1121,7 @@ def copy(self, name=None):
         view.toolbar.configuration.item_select('Copy this Role to a new Role')
         view = self.create_view(AddRoleView, wait=10)  # implicit assert
         new_role = self.parent.instantiate(name=name)
-        view.fill({'name_txt': new_role.name})
+        view.fill({'name_txt': new_role.name, "vm_restriction_select": restriction})
```
Comment (kedark3) [inline cfme/configure/access_control/__init__.py]: **QUESTION:** Is it valid to pass `None` for restriction? if not, we might want to add a check.

File: cfme/tests/ssui/test_ssui_myservice.py [hunk 3]
```diff
@@ -29,14 +30,70 @@
...
+            required_fields=['provisioning'])]),
+    pytest.mark.usefixtures("setup_provider")
 ]
 
 
+@pytest.fixture(scope='module')
+def new_role(appliance):
+    role = appliance.collections.roles.instantiate(name='EvmRole-user_self_service')
+    copied_role = role.copy(name="{role}_{name}".format(
+        role=role.name, name=fauxfactory.gen_alpha()), restriction="None")
```
Comment (kedark3) [inline cfme/tests/ssui/test_ssui_myservice.py]: **QUESTION**
Above https://github.com/ManageIQ/integration_tests/pull/9036/files#diff-4f9b3a3940e709873c58aa1fdf9b95ccR1110 you are passing `None<NoneType>` while here you pass it as `"None"<str>` 
Is there a reason behind this?

File: cfme/tests/ssui/test_ssui_myservice.py [hunk 4]
```diff
@@ -29,14 +30,70 @@
...
+        email=fauxfactory.gen_email(),
+        groups=new_group,
+        cost_center='Workload',
+        value_assign='Database')
+    yield user
+    user.delete_if_exists()
+
+
+@pytest.mark.meta(automates=[1699943])
+@pytest.mark.ignore_stream('5.10')
```
Comment (sshveta) [inline cfme/tests/ssui/test_ssui_myservice.py]: Question : Why is this skipped for 5.10 ? 
The BZ is for 5.11

File: cfme/tests/ssui/test_ssui_myservice.py [hunk 5]
```diff
@@ -29,14 +30,70 @@
...
+    collection = appliance.collections.groups
+    group = collection.create(description='group_{}'.format(fauxfactory.gen_alphanumeric()),
+                              role=new_role.name,
+                              tenant=appliance.collections.tenants.get_root_tenant().name)
+    yield group
+    group.delete_if_exists()
+
+
+@pytest.fixture(scope='module')
+def new_user(appliance, new_group, new_credential):
```
Comment (sshveta) [inline cfme/tests/ssui/test_ssui_myservice.py]: I see fixtures of new_user , new_group and new_role used in lot of tests.
Let's create fixture for these in common place .

**Comment by dajoRH** [comment]
I detected some fixture changes in commit 2c15a63d8cacf4f1504d7958dbbf27e403338e1d

The *local* fixture **`new_role`** is used in the following files:
- **cfme/tests/ssui/test_ssui_myservice.py**
    - *new_group*

The *local* fixture **`new_group`** is used in the following files:
- **cfme/tests/ssui/test_ssui_myservice.py**
    - *new_user*

The *local* fixture **`new_user`** is used in the following files:
- **cfme/tests/ssui/test_ssui_myservice.py**
    - *test_retire_service_ssui_user*

Please, consider creating a PRT run against these tests make sure your fixture changes do not break existing usage :smiley:

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by niyazRedhat** [comment]
Closing the pr. Nikhil automated this pr with more sneario 9620

**Pull request closed without merging**